src/js/entities-service/entities/patients.js

```diff
@@ -35,22 +35,27 @@ const _Model = BaseModel.extend({
   addGroup(group) {
     const groups = this.get('_groups') || [];
 
-    this.set('_groups', groups.concat(group.id));
+    this.set('_groups', groups.concat({
+      id: group.id,
+    }));
   },
-  removeGroup(group) {
+  removeGroup(removedGroup) {
     const groups = this.get('_groups');
 
-    this.set('_groups', reject(groups, groupId => groupId === group.id));
+    this.set('_groups', reject(groups, group => group.id === removedGroup.id));
```

Comment:
This was broken, the format of `removedGroup` was incorrect, so nothing would ever get rejected.